contracts/cryptoblades.sol

```diff
@@ -711,20 +711,25 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         }
     }*/
 
-    function mintWeaponN(uint32 num, uint16 chosenElement, uint16 chosenElementFee)
+    function mintWeaponN(uint32 num, uint8 chosenElement)
```

Comment:
I just noticed and I think we should only allow passing 0, 1, 2, 3, 100 as chosenElement, other values are not supported and could cause unexpected behavior.
Please create a modifier to check for supported elements and apply to both mint functions.
Something along those lines:
```
modifier isSupportedElement(uint8 element) {
    _isSupportedElement(element);
    _;
}

function _isSupportedElement(uint8 element) internal view {
   require(element == 100 || (element>= 0 && element<= 3), "Not supported element");
}
```

I'm not too sure what effect it will have on the gas cost - if it's significant it might be better to create a separate function for mints with chosenElement and leave this as it was originally for random elements. @poshdan please share your opinion on this.

Reply:
gas costs are less important for things like minting - daily we have 3 million fight transactions but weapon mints are only in the thousands (let's say 50k weapons made, and that's not accounting for how many weapons are made via 10x).

That would make once cent saved on fights 60x more important than one cent saved in mints.
Of course there's no need to be wasteful, but I'm not going to lose sleep over either approach.
The tradeoff with two functions instead of one, is that the contract becomes larger, and we are indeed close to hitting limits. We can switch it around later if we need to compress the contract though.

Feel free to try the function before and after in your local environment or run tuffle console's estimate function like so:
mintWeapon.estimateGas(parameters of mintWeapon). Multiply whatever gas cost you get by x1.45 (which is gonna be roughly its cost on mainnet), and at 5 gwei it takes 200 million gas per BNB. If the difference in gas is less than several thousand, it's not a big deal.

Reply:
With the modifier:
![image](https://user-images.githubusercontent.com/87876521/129213101-f37cd1af-71cc-466a-bd04-bd34a6d109c3.png)

With the separate functions:
![image](https://user-images.githubusercontent.com/87876521/129213251-72fca385-f71c-4267-ae16-7e66ba48c7d2.png)

I personally favor the modifier solution because its much less code and thus easier to read but you guys decide.

Reply:
Nice, go for the modifier definitely then. Thanks!

Reply:
Alright updated it thanks for your help guys!